### PR TITLE
[BTAT-7558] - confirmation message for digital preference email verif…

### PR DESCRIPTION
**Pull request by sraspin**
…ied users

# Pull Request Checklist
* [ ] Branch is rebased with master
* [ ] Added Unit Tests for changed functionality
* [ ] Ran `sbt clean compile coverage test it:test coverageReport`, all tests pass and coverage meets minimum
* [ ] Checked libraries are up to date with `sbt validate`. (If applicable) library updates are done in a separate, single commit
* [ ] Ran Acceptance tests as per the story. If any query ping in group slack and confirm the AT's repo
* [ ] Ran Performance tests as per the story. If any query ping in group slack and confirm the PT's repo
* [ ] (If applicable) raised app-config-* PRs
* [ ] (If applicable) ran Wave test for changes to views
* [ ] (If applicable) added Integration Test

### CODE REVIEW

File: test/controllers/BusinessAddressControllerSpec.scala [hunk 1]
```diff
@@ -346,43 +348,80 @@ class BusinessAddressControllerSpec extends ControllerBaseSpec with MockAddressL
...
+
+          "return HTML" in {
+            contentType(result) shouldBe Some("text/html")
+            charset(result) shouldBe Some("utf-8")
+          }
+
+          "render the Business Address confirmation view" in {
+            messages(document.select("h1").text) shouldBe ChangeAddressConfirmationPageMessages.heading
+            messages(document.select("#content article p:nth-of-type(1)").text()) shouldBe ChangeAddressConfirmationPageMessages.digitalPref
+          }
```
Comment (miranda-hawkes) [inline test/controllers/BusinessAddressControllerSpec.scala]: Should add a test for when call to `customerCircumstanceDetailsService.getCustomerCircumstanceDetails` fails (and it defaults it to false)

File: app/controllers/BusinessAddressController.scala [hunk 2]
```diff
@@ -90,17 +90,20 @@ class BusinessAddressController @Inject()(val authenticate: AuthPredicate,
...
+      cPref <- contactPreferenceService.getContactPreference(user.vrn) map {
+        case Right(contact) =>
+          auditService.extendedAudit(
+            ContactPreferenceAuditModel(user.vrn, contact.preference, ContactPreferenceAuditKeys.changeBusinessAddressAction),
+            Some(controllers.routes.ChangeBusinessNameController.show().url)
+          )
+          Some(contact.preference)
+        case _ => None
+      }
+      verifiedEmail <- customerCircumstanceDetailsService.getCustomerCircumstanceDetails(user.vrn) map {
```
Comment (miranda-hawkes) [inline app/controllers/BusinessAddressController.scala]: You could add logic here so it only calls off to the service if the contact preference is digital, therefore reducing calls to DES where the data isn't used

**Comment by sraspin** [comment]
https://github.com/hmrc/app-config-base/pull/3140
https://github.com/hmrc/app-config-production/pull/5904

File: app/config/features/Features.scala [hunk 3]
```diff
@@ -32,4 +32,5 @@ class Features @Inject()(implicit config: Configuration) {
   val useLanguageSelector = new Feature(ConfigKeys.useLanguageSelectorFeature)
   val useOverseasIndicator = new Feature(ConfigKeys.useOverseasIndicator)
   val changeClientFeature = new Feature(ConfigKeys.changeClientFeature)
+  val emailVerifiedFeature = new Feature(ConfigKeys.emailVerifiedFeature)
```
Comment (miranda-hawkes) [inline app/config/features/Features.scala]: Need to add the new feature to the feature switch view and controller to make it toggle-able :)

File: test/controllers/BusinessAddressControllerSpec.scala [hunk 4]
```diff
@@ -346,43 +348,161 @@ class BusinessAddressControllerSpec extends ControllerBaseSpec with MockAddressL
...
+        "emailVerified feature is disabled" should {
 
-          verify(mockAuditingService)
-            .extendedAudit(
-              ArgumentMatchers.any[ContactPreferenceAuditModel],
-              ArgumentMatchers.any[Option[String]]
+          lazy val result = {
+            mockConfig.features.emailVerifiedFeature(false)
+            mockContactPreferenceSuccess(ContactPreference("DIGITAL"))
+            mockCustomerDetailsSuccess(customerInformationModelMaxOrganisation)
```
Comment (miranda-hawkes) [inline test/controllers/BusinessAddressControllerSpec.scala]: I'd expect this service to not be called at all in this scenario so you can remove the mock

**Pull request merged**